### ISSUE TITLE
Setting seed does not work

### ISSUE THREAD

**Issue by hust-nj** [graphdeeplearning/benchmarking-gnns#7]
Hi, thank you for sharing the project with us.
I'm confused about that though I **fixed the random seed** as your config file did, I still got different results between different runs just like without setting seed. 
I want to know that if fixing the seed can ensure the reproducibility and get the same accuracy?

**Comment by chaitjo** [comment]
Thanks for your interest, @hust-nj. Maybe you can elaborate if you want help? At which decimal place were the results different, what was your experimental setup, etc?

In my experience, there should not be any big change in performance across different runs with a fixed seed. If that is the case, we'd like to investigate further...

In general, we took care to implement fixing random seeds for reproducibility, following best practices, e.g. https://pytorch.org/docs/stable/notes/randomness.html. However, there may still be some element of randomness/non-determinism we couldn't account for.

**Comment by chaitjo** [comment]
Closing due to no follow up.

**Comment by socoolzjm** [comment]
There is a bug in setting the random seed code.

` if device == 'cuda':` will never be True. I fixed it by adding .type `if device.type == 'cuda':`

I also added `torch.backends.cudnn.deterministic = True` and `torch.backends.cudnn.benchmark = False` but still could not reproduce the results in paper.

For ZINC, I got:
GCN: mean: 0.469275, s.d. 0.007346370986
GAT: mean: 0.45945, s.d. 0.007597587336
In the paper:
GCN: mean: 0.469, s.d. 0.002
GAT: mean: 0.463, s.d. 0.002

I don't know why my standard deviation results are lager
